include/boost/beast/http/basic_file_body.hpp

```diff
@@ -210,8 +223,29 @@ reset(File&& file, error_code& ec)
 
     // Cache the size
     file_size_ = file_.size(ec);
+
+    // Consider the offset
+    if (!ec)
+        file_size_ -= file_.pos(ec);
+}
+
+template<class File>
+void
+basic_file_body<File>::
+value_type::
+seek(std::uint64_t offset, error_code& ec)
+{
+    file_.seek(offset, ec);
```

Comment:
In my testing of Beast with my own implementation of file serving and seeking from the system, reading the last part of a file will not only set the eof bit (which seeking automatically clears) but also the fail bit. So before seeking here I think you might want to test for failure / error and if so, clear the fail bit condition.

Though this is only if you want to make the file class here be more stateful with general seeking.  But if you were to only allow seeking the file once (say through an overload of the reset method and getting rid of the seek method) then you don't need to test for this condition. If the user wants to seek the file again then they need to reset it or discard the current object and instantiate a new file class object.